ocaml/xapi/xapi_host_driver.ml

```diff
@@ -0,0 +1,160 @@
+(*
+   Copyright (c) Cloud Software Group, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation; version 2.1 only. with the special
+   exception on linking described in file LICENSE.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+ *)
+
+module D = Debug.Make (struct let name = __MODULE__ end)
+
+open D
+module Unixext = Xapi_stdext_unix.Unixext
+
+(*
+module DriverMap = Map.Make (String)
+module DriverSet = Set.Make (String)
+*)
+module T = Xapi_host_driver_tool
+
+let invalid_value field value =
+  raise Api_errors.(Server_error (invalid_value, [field; value]))
+
+let internal_error fmt =
+  Printf.ksprintf
+    (fun msg ->
+      error "%s" msg ;
+      raise Api_errors.(Server_error (internal_error, [msg]))
+    )
+    fmt
+
+let drivertool args =
+  let path = !Xapi_globs.driver_tool in
+  try
+    let stdout, _stderr = Forkhelpers.execute_command_get_output path args in
```

Comment:
Do we want to log the stderr as well?

Reply:
On the error path stderr is not available. So we either succeed (where stderr should be meaningless) or we fail. For now, I'd keep it like this.